src/modules/musig/keyagg_impl.h

```diff
@@ -0,0 +1,293 @@
+/***********************************************************************
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#ifndef SECP256K1_MODULE_MUSIG_KEYAGG_IMPL_H
+#define SECP256K1_MODULE_MUSIG_KEYAGG_IMPL_H
+
+#include <string.h>
+
+#include "keyagg.h"
+#include "../../eckey.h"
+#include "../../ecmult.h"
+#include "../../field.h"
+#include "../../group.h"
+#include "../../hash.h"
+#include "../../util.h"
+
+static const unsigned char secp256k1_musig_keyagg_cache_magic[4] = { 0xf4, 0xad, 0xbb, 0xdf };
+
+/* A keyagg cache consists of
+ * - 4 byte magic set during initialization to allow detecting an uninitialized
+ *   object.
+ * - 64 byte aggregate (and potentially tweaked) public key
+ * - 64 byte "second" public key (set to the point at infinity if not present)
+ * - 32 byte hash of all public keys
+ * - 1 byte the parity of the internal key (if tweaked, otherwise 0)
+ * - 32 byte tweak
+ */
+/* Requires that cache_i->pk is not infinity. */
+static void secp256k1_keyagg_cache_save(secp256k1_musig_keyagg_cache *cache, secp256k1_keyagg_cache_internal *cache_i) {
```

Comment:
in https://github.com/bitcoin-core/secp256k1/commit/f8f4f3c7a10345295ed72cdae0faecff37a919e5:

Any reason `*cache_i` isn't const? You'd need to also the in param for both `_ge_to_bytes` and `_ge_to_bytes_ext` to be const, but seems like they should be unless I'm missing something.

Reply:
FWIW, I changed these functions to `const` and was able to compile and run the tests.

Reply:
Thanks, fixed.